docs/rpc-endpoints.md

```diff
@@ -0,0 +1,233 @@
+
+```
+{
+  "functions": [
+    {
+      "name": "exotic-block-height",
+      "access": "private",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": "bool"
+      }
+    },
+    {
+      "name": "update-info",
+      "access": "public",
+      "args": [],
+      "outputs": {
+        "type": {
+          "response": {
+            "ok": "bool",
+            "error": "none"
+          }
+        }
+      }
+    },
+    {
+      "name": "get-exotic-data-info",
+      "access": "read_only",
+      "args": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "outputs": {
+        "type": {
+          "tuple": [
+            {
+              "name": "btc-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "burn-block-time",
+              "type": "uint128"
+            },
+            {
+              "name": "id-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-hash",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            },
+            {
+              "name": "stacks-miner",
+              "type": "principal"
+            },
+            {
+              "name": "vrf-seed",
+              "type": {
+                "buffer": {
+                  "length": 32
+                }
+              }
+            }
+          ]
+        }
+      }
+    }
+  ],
+  "variables": [],
+  "maps": [
+    {
+      "name": "block-data",
+      "key": [
+        {
+          "name": "height",
+          "type": "uint128"
+        }
+      ],
+      "value": [
+        {
+          "name": "btc-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "burn-block-time",
+          "type": "uint128"
+        },
+        {
+          "name": "id-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-hash",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        },
+        {
+          "name": "stacks-miner",
+          "type": "principal"
+        },
+        {
+          "name": "vrf-seed",
+          "type": {
+            "buffer": {
+              "length": 32
+            }
+          }
+        }
+      ]
+    }
+  ],
+  "fungible_tokens": [],
+  "non_fungible_tokens": []
+}
+```
+
+### GET /v2/contracts/source/[Stacks Address]/[Contract Name]
+
+Fetch the source for a smart contract. Returned as a JSON string.
```

Comment:
Shouldn't this just be `text/plain` and serve the _raw_ Clarity source?

Reply:
I think it's better for all of these endpoints to return JSON.

Reply:
So what does this endpoint's HTTP body look like?  If it's raw Clarity code, it won't parse as JSON.

Reply:
It’ll be a JSON escaped string.

Reply:
Gotcha.